File: Python/Encapsulamiento/cubo.py
class Cubo:
    def __init__(self, longitud_lado):
        """Inicializa el objeto Cubo e indica los valores de los atributos"""
        if longitud_lado <= 0:
            return ValueError("El valor de la longitud lado debe ser mayor a 0")
        
        self.longitud_lado = longitud_lado
        self.caras = 6
        self.area = pow(self.longitud_lado, 2) * self.caras
        self.volumen = pow(self.longitud_lado, 3)

    def obtener_lado(self):
        """Devuelve el lado del cubo"""
        return self.longitud_lado
    

    def cambiar_lado(self, longitud_lado):
        """Cambia el lado del cubo"""
        if (longitud_lado <= 0):
            return ValueError("El valor de la longitud lado debe ser mayor a 0")
        
        self.longitud_lado = longitud_lado
    
    def obtener_area(self):
        """Devuelve el area del cubo"""
        return self.area
    
    def cambiar_area_cara(self, area_cara):
        """Cambia el area del cubo"""
        if area_cara <= 0:
            return ValueError("El valor del area debe ser mayor a 0")
        
        self.area = area_cara
    
    def obtener_volumen(self):
        """Devuelve el volumen del cubo"""
        return self.volumen
    
    def cambiar_volumen(self, volumen):
        """Cambia el volumen del cubo"""
        if (volumen <= 0):
            return ValueError("El valor del volumen debe ser mayor a 0")
        
        self.volumen = volumen

cubo = Cubo(12)
print(cubo.obtener_lado())
print(cubo.obtener_area())

File: Java/src/Encapsulamiento/Cubo.java
package Encapsulamiento;

class Cubo {

    private double longitudLado;

    /**
     * pre : longitudLado es un valor mayor a 0.
     * post: inicializa el cubo a partir de la longitud de lado dada
     */
    public Cubo (double longitudLado) { 
        if (longitudLado > 0) {
            this.longitudLado = longitudLado;
        } else {
            System.out.println(new Error("La longitud debe ser mayor a 0"));
        }
    }

    /**
     * post: devuelve la longitud de todos los lados del cubo
     */
    public double obtenerLado() {
        return longitudLado;
    }

    /**
     * pre : longitudLado es un valor mayor a 0.
     * post: cambia la longitud de todos los lados del cubo
     */
    public void cambiarLado(double longitudLado) {
        if (longitudLado > 0) {
            this.longitudLado = longitudLado;
        } else {
            System.out.println(new Error("La longitud del lado debe ser mayor a 0"));
        }
    }

    /**
     * post: devuelve el área de la superficie de las caras del cubo
     */
    public double obtenerAreaCara() {
        return 6 * Math.pow(longitudLado, 2);
    }

    /**
     * pre: areaCara es un valor mayor a 0.
     * post: cambia el área de las caras del cubo
     */
    public void cambiarAreaCara(double areaCara) {
        if (areaCara > 0) {
            this.longitudLado = Math.sqrt(areaCara / 6);
        }
    }

    /**
     * post: devuelve el volumen que encierra el cubo
     */
    public double obtenerVolumen() {
        return Math.pow(longitudLado, 3);
    }

    /**
     * pre: volumen es un valor mayor a 0.
     * post: cambia el volumen del cubo
     */
    public void cambiarVolumen(double volumen) {
        if (volumen > 0) {
            this.longitudLado = Math.pow(volumen, 1.0 / 3);
        }
    }

    public static void main(String[] args) {
        Cubo cubo = new Cubo(5);

        System.out.println("Lado cubo: " + cubo.obtenerLado());

        System.out.println("Area cubo: " + cubo.obtenerAreaCara());

        System.out.println("Volumen cubo: " + cubo.obtenerVolumen());

        cubo.cambiarLado(10);

        cubo.cambiarAreaCara(12);

        cubo.cambiarVolumen(20);

        System.out.println("Lado cubo: " + cubo.obtenerLado());

        System.out.println("Area cubo: " + cubo.obtenerAreaCara());

        System.out.println("Volumen cubo: " + cubo.obtenerVolumen());

        
    }
}